src/components/TextInput/index.android.js

```diff
@@ -0,0 +1,25 @@
+import React, {forwardRef} from 'react';
+import styles from '../../styles/styles';
+import BaseTextInput from './BaseTextInput';
+import * as baseTextInputPropTypes from './baseTextInputPropTypes';
+import CONST from '../../CONST';
+
+const TextInput = forwardRef((props, ref) => (
+    <BaseTextInput
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...props}
+
+        // Setting autoCompleteType to new-password throws an error on Android/iOS, so fall back to password in that case
+        // eslint-disable-next-line react/jsx-props-no-multi-spaces
+        autoCompleteType={props.autoCompleteType === 'new-password' ? 'password' : props.autoCompleteType}
+        innerRef={ref}
+        inputStyle={[styles.baseTextInput, ...props.inputStyle]}
+        getKeyboardType={passwordHidden => (props.secureTextEntry && !passwordHidden ? CONST.KEYBOARD_TYPE.VISIBLE_PASSWORD : props.keyboardType)}
```

Comment:
@b1tjoy Your change works but can we DRY up some code here? It seems the only extra prop that we're using is `getKeyboardType`. 

1. So instead of have duplicated code for `TextInput/index.ios.js` and `TextInput/index.android.js`, we retain `TextInput/index.native.js`

2. We can create a function `keyboardType={getKeyboardType(all_there_params_you need)}`

3. Copy your current conditions to an android specific function `getKeyboardType/index.android.js`

4. and for other platforms `getKeyboardType` a  no op.

This will ensure less duplication of code between android and ios `TextInput` files